app/js/enhancedDate.js

```diff
@@ -1,5 +1,133 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function (date) {
+// still maintains access to all globals
+  
+  var setDate = {};
+
+//Stores the passed-in date.  
+  setDate = function (date) {
+      var myDate = date;
+    
+//Do some type-checking to make sure you have a valid type.    
+      if(typeof myDate !== null)
+      {
+//Takes milliseconds after epoch or a date object. 
+        
+//Do some type-checking to make sure you have a valid type. 
+        if (typeof myDate == Date() || typeof myDate == "number"){
+          myDate = date;
```

Comment:
This works, but if someone modifies the Date object they passed in outside of your module, it will modify it inside your module as well because objects are passed by reference. To get around this, set myDate by just passing in the date to the Date constructor: var myDate = new Date(date); It works for Date objects or millisecond values.